### ISSUE TITLE
Operand of null-aware operation '?.' has type 'SchedulerBinding'

### ISSUE THREAD

**Issue by ankitparmar007** [Baseflow/flutter_cached_network_image#732]
## 🐛 Bug Report
**_Warning After upgrading flutter to 3.0.0:_**

```
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/cached_network_image-3.2.0/lib/src/image_provider/multi_image_stream_completer.dart:152:22: Warning: Operand of null-aware operation '?.' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/scheduler/binding.dart').
    SchedulerBinding.instance?.scheduleFrameCallback(_handleAppFrame);
                     ^
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/cached_network_image-3.2.0/lib/src/image_provider/cached_network_image_provider.dart:109:29: Warning: Operand of null-aware operation '?.' has type 'PaintingBinding' which excludes null.
 - 'PaintingBinding' is from 'package:flutter/src/painting/binding.dart' ('/C:/src/flutter/packages/flutter/lib/src/painting/binding.dart').
      () => PaintingBinding.instance?.imageCache?.evict(key),
                            ^
```

**Comment by MindMayhem** [comment]
https://github.com/imaNNeoFighT/fl_chart/issues/997
https://github.com/Milad-Akarie/auto_route_library/issues/1065

**Comment by beaurepair** [comment]
https://github.com/Baseflow/flutter_cached_network_image/pull/727

**Comment by C1rdec** [comment]
https://github.com/Baseflow/flutter_cached_network_image/pull/734

**Comment by noga-dev** [comment]
Add this to pubspec for now:

```yaml
dependency_overrides:
  cached_network_image:
    git:
      url: https://github.com/XuannThucc/flutter_cached_network_image.git
      path: cached_network_image
      ref: ff72f00f142f13d889d7549e013af91cb0b523ab
```

**Comment by andreclinio** [comment]
As somebody solved the problem (as @noga-dev pointed above), I think the authors will release a patch soon.
Any perspective from the authros?

**Comment by neoacevedo** [comment]
This happens too with Flutter 2.17.

**Comment by RoyalCoder88** [comment]
have the same problem :( any updates? thanks!

**Comment by nerder** [comment]
> Add this to pubspec for now:
> 
> ```yaml
> dependency_overrides:
>   cached_network_image:
>     git:
>       url: https://github.com/XuannThucc/flutter_cached_network_image.git
>       path: cached_network_image
>       ref: ff72f00f142f13d889d7549e013af91cb0b523ab
> ```

Not a good idea to link to an "untrusted" fork. In order to fix this please point to your own fork, or eventually to a PR in this repo.

Please @renefloor take a look at this when you can.

**Comment by noga-dev** [comment]
> **[nerder](/nerder) ** commented [16 minutes ago](#issuecomment-1126718514)

It's fine. And the ref tag in the above code snippet is there for the extra paranoid reasons you must be pondering about.

**Comment by nerder** [comment]
Thanks for the clarification @noga-dev, better being paranoid than sorry I just saw that I have to go back and 😄 

Anyway would be nice if this gets merged by @renefloor

**Comment by renefloor** [comment]
Update is released as cached_network_image: ^3.2.1
🚀

Btw, these warnings are just warnings. Very annoying, but still compiling and running fine. Don't bother other maintainers too much, they for sure know that they need to update this.
